### PR TITLE
Binance: Fix websocket subscribe error with many currency pairs enabled

### PR DESCRIPTION
**Pull request by arttobe**
# PR Description

It is now possible to subscribe to many currencies, before it was limited to 4096 bytes and an error was thrown
Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Travis with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by MadCozBadd — APPROVED** [review]
tACK! Awesome fix!!!!!! Thank you so much for your contribution

Just a note incase anyone encounters the following issue:
@gloriousCode helped me figure out that there is still an issue when we subscribe all the available pairs since 1 connection can only handle 1024 streams. And Binance has about 1007 available pairs so having all of them connect via ticker, ob, trades... causes binance to boot off that connection. But that is not your issue. We will handle it soon!

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thank you for adding this! Really appreciated. I've tested this and it works! Just one comment around the rate limit use

File: exchanges/binance/binance_websocket.go [hunk 1]
```diff
@@ -540,14 +540,23 @@ func (b *Binance) Subscribe(channelsToSubscribe []stream.ChannelSubscription) er
 	payload := WsPayload{
 		Method: "SUBSCRIBE",
 	}
-
 	for i := range channelsToSubscribe {
 		payload.Params = append(payload.Params, channelsToSubscribe[i].Channel)
+		if i%50 == 0 && i != 0 {
+			time.Sleep(time.Second)
```
[the remark this answers is not in the document]
Comment (arttobe) [reply exchanges/binance/binance_websocket.go:546]: Where did i find the property RateLime in b.Websocket?

File: exchanges/binance/binance_websocket.go [hunk 2]
```diff
@@ -540,14 +540,23 @@ func (b *Binance) Subscribe(channelsToSubscribe []stream.ChannelSubscription) er
...
+			err := b.Websocket.Conn.SendJSONMessage(payload)
+			if err != nil {
+				return err
+			}
+			payload.Params = []string{}
+		}
+
 	}
 	err := b.Websocket.Conn.SendJSONMessage(payload)
```
Comment (woshidama323) [inline exchanges/binance/binance_websocket.go:555]: Here binance still had disconnected error log on my testing env , Is that a problem if the payload get empty

Comment (MadCozBadd) [reply exchanges/binance/binance_websocket.go:555]: Thanks for bringing up the issue! However would you be able to elaborate a little more so we can reproduce this scenario

File: exchanges/binance/binance_websocket.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/binance/binance_websocket.go:546]: `RateLimit` is a property of `WebsocketConnection` and not `b.Websocket`. To set a ratelimit, look at the `binance_wrapper.go` file and look for the code `b.Websocket.SetupNewConnection`. From there, you can add the property `RateLimit` to the `stream.ConnectionSetup` struct like in the above code snippet.

**Review by nitsuaaisblink — APPROVED** [review]

**Review by shazbert — CHANGES_REQUESTED** [review]
Could we please add this update to the unsubscribe method as well. Waiting on the integration with the rate limit system if possible. Then it seems good to go. Thank you very much for opening this PR.

**Review by shazbert — APPROVED** [review]
tested approved.

**Review by MadCozBadd — APPROVED** [review]
tACK! Awesome addition! thanks @arttobe & @gloriousCode

**Review by thrasher- — APPROVED** [review]
tACK, but with the caveat already mentioned

**Pull request merged**